src/sources/logplex.rs

```diff
@@ -147,7 +147,14 @@ fn line_to_event(line: String) -> Event {
             fields = parts.len()
         );
         Event::from(line)
-    }
+    };
+
+    // Add source type
+    event
+        .as_mut_log()
+        .try_insert(event::log_schema().source_type_key(), "logplex");
```

Comment:
Having a two-parameter `Event::from` would eliminate this need for a `mut event`.

Reply:
It is two parameter at the moment, but we would need to add a third one for `source_type_key` in the near future, so I don't think having a specialized `from` would save us a lot.

Reply:
The same goes for specializing `LogEvent::new` and `Event::new_empty_log`.